File: CS/CS/Controllers/HomeController.cs
using System;
using System.Web;
using System.Web.Mvc;
using CS.Models;

namespace CS.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            if(Session["DataTableModel"] == null)
                Session["DataTableModel"] = InMemoryModel.OpenExcelFile();

            return View(Session["DataTableModel"]);
        }
        public ActionResult DataTableDataBindingPartial() {
            return PartialView(Session["DataTableModel"]);
        }
    }
}

File: VB/CS/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports CS.Models

Namespace CS.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            If Session("DataTableModel") Is Nothing Then Session("DataTableModel") = InMemoryModel.OpenExcelFile()
            Return View(Session("DataTableModel"))
        End Function

        Public Function DataTableDataBindingPartial() As ActionResult
            Return PartialView(Session("DataTableModel"))
        End Function
    End Class
End Namespace
